### DISCUSSION TITLE
Bypass certificate validition for a specific domain

### DISCUSSION THREAD

**Discussion by Lucadaff** [mitmproxy/mitmproxy#5240] [General]
Is there a way to bypass the certificate validation for a specific domain without specifying the option `ssl_insecure` at the start?
I inherit a code containing a modification to mitmproxy's source code (version 5.1.1) in the file connections.py, class ServerConnection, method establish_tls. This modification allows to switch to insecure mode, by setting a custom variable in the mitmproxy event if some domain is matched.

    def establish_tls(self, *, sni=None, client_certs=None, **kwargs):
        ...
        if self.my_insecure == True:
            if kwargs["verify"] != SSL.VERIFY_NONE:
                kwargs["verify"] = SSL.VERIFY_NONE

Now I upgrade to mitmproxy 8.0.0, I have noticed that to replicate the mitmproxy's source code modification I have to modify the files connections.py and addons/tlsconfig.py. I would like to know if I can obtain the same result in a simplier way.

**Comment by mhils** [comment]
We don't have an option for this at the moment, but I'd be happy to review a PR that adds one. :)
